File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/models/RobotFirmware.java
package com.neatorobotics.sdk.android.models;

import java.io.Serializable;

/**
 * RobotFirmware model
 */
public class RobotFirmware implements Serializable {
    private static final String TAG = "RobotFirmware";

    public String model, version, url, changelogurl;
    public int filesize;

    //costructors
    public RobotFirmware(){}
    public RobotFirmware(String model, String version, String url, int filesize, String changelogurl){
        this.model = model;
        this.version = version;
        this.url = url;
        this.filesize = filesize;
        this.changelogurl = changelogurl;
    }
}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/models/RobotFirmware.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.models

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class RobotFirmware(var model: String? = null,
                    var version: String? = null,
                    var url: String? = null,
                    var manualUpdateInfoUrl: String? = null,
                    var filesize: Int = 0,
                    var minRequiredVersion: String? = null) : Parcelable
